app/modules/asset/service.server.ts

```diff
@@ -540,6 +540,46 @@ export async function deleteNote({
   });
 }
 
+export async function duplicateAsset({
+  asset,
+  userId,
+  amountOfDuplicates,
+}: {
+  asset: Prisma.AssetGetPayload<{ include: { custody: true } }>;
+  userId: string;
+  amountOfDuplicates: number;
+}) {
+  const duplicatedAssets = [];
+
+  for (let i = 0; i < amountOfDuplicates; i++) {
+    const duplicatedAsset = await db.asset.create({
```

Comment:
Here you are making it hard on yourself and also a bit more difficult to maintain. We already have a function called `createAsset` in the same file that handles all the cases and logic. You should just use that function.

Reply:
If I use `createAsset` then I will not be able to duplicate the image of the asset.
If I want to use `createAsset` I have to accept two more args to do the job

Reply:
also any reason for creating 1 by 1, may be creating all at once using `createMany` is more performant.

Reply:
> also any reason for creating 1 by 1, may be creating all at once using `createMany` is more performant.

It is for sure more performant but we have some complexity related to duplicating the images which will make it very convoluted if we do createMany and the performance gains from this I dont see valuable enough to sacrifice DX and having to create a way more complex code to handle this.